cmd/opentelemetry-collector/cmd/all-in-one/main.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/elasticsearch"
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/memory"
+	queryApp "github.com/jaegertracing/jaeger/cmd/query/app"
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	jConfig "github.com/jaegertracing/jaeger/pkg/config"
+	"github.com/jaegertracing/jaeger/pkg/version"
+	pluginStorage "github.com/jaegertracing/jaeger/plugin/storage"
+	cassStorage "github.com/jaegertracing/jaeger/plugin/storage/cassandra"
+	esStorage "github.com/jaegertracing/jaeger/plugin/storage/es"
+	"github.com/jaegertracing/jaeger/storage"
+)
+
+func main() {
+	handleErr := func(err error) {
+		if err != nil {
+			log.Fatalf("Failed to run the service: %v", err)
+		}
+	}
+
+	ver := version.Get()
+	info := service.ApplicationStartInfo{
+		ExeName:  "jaeger-opentelemetry-all-in-one",
+		LongName: "Jaeger OpenTelemetry all-in-one",
+		Version:  ver.GitVersion,
+		GitHash:  ver.GitCommit,
+	}
+
+	v := viper.New()
+	storageType := os.Getenv(pluginStorage.SpanStorageTypeEnvVar)
+	if storageType == "" {
+		storageType = "memory"
+	}
+
+	configCreatedWait := sync.WaitGroup{}
+	configCreatedWait.Add(1)
+	exporters := configmodels.Exporters{}
+
+	cmpts := defaults.Components(v)
+	cfgFactory := func(otelViper *viper.Viper, f config.Factories) (*configmodels.Config, error) {
+		collectorOpts := &collectorApp.CollectorOptions{}
+		collectorOpts.InitFromViper(v)
+		cfgOpts := defaults.ComponentSettings{
+			ComponentType:  defaults.AllInOne,
+			Factories:      cmpts,
+			StorageType:    storageType,
+			ZipkinHostPort: collectorOpts.CollectorZipkinHTTPHostPort,
+		}
+		cfg, err := cfgOpts.CreateDefaultConfig()
+		if err != nil {
+			return nil, err
+		}
+
+		if len(builder.GetConfigFile()) > 0 {
+			otelCfg, err := service.FileLoaderConfigFactory(otelViper, f)
+			if err != nil {
+				return nil, err
+			}
+			err = defaults.MergeConfigs(cfg, otelCfg)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		exporters = cfg.Exporters
+		configCreatedWait.Done()
+		return cfg, nil
+	}
+
+	svc, err := service.New(service.Parameters{
+		ApplicationStartInfo: info,
+		Factories:            cmpts,
+		ConfigFactory:        cfgFactory,
+	})
+	handleErr(err)
+
+	// Add Jaeger specific flags to service command
+	// this passes flag values to viper.
+	storageFlags, err := app.AddStorageFlags(storageType, true)
+	if err != nil {
+		handleErr(err)
+	}
+	cmd := svc.Command()
+	jConfig.AddFlags(v,
+		cmd,
+		app.AddComponentFlags,
+		storageFlags,
+		queryApp.AddFlags,
+	)
+
+	// parse flags to propagate Jaeger config file flag value to viper
+	cmd.ParseFlags(os.Args)
+	err = jflags.TryLoadConfigFile(v)
+	if err != nil {
+		handleErr(fmt.Errorf("could not load Jaeger configuration file %w", err))
+	}
+
+	go func() {
+		err = svc.Start()
+		handleErr(err)
+	}()
+
+	configCreatedWait.Wait()
+	exp := getStorageExporter(storageType, exporters)
```

Comment:
What if a comma separated list of storage types are provided?

Reply:
That does not seem like that somebody would use multiple storages with all-in-one. We can default to the first one in the list.

That said I would remove the support for the list that can be done via OTEL config.